api-helpers/contentCreatorFunctions.ts

```diff
@@ -0,0 +1,152 @@
+import { resolve } from 'path';
+import { URL } from 'url';
+
+import Boom from '@hapi/boom';
+import cheerio from 'cheerio';
+import Slugify from 'slugify';
+
+import { closeConnection, openConnection } from './db';
+
+const NEVER = new Date(0);
+
+type BlogData = {
+  readonly name: string;
+  readonly href: string;
+  readonly favicon: string;
+  readonly feeds: readonly Feed[];
+};
+
+type Feed = {
+  readonly type: string;
+  readonly url: string;
+  readonly title: string;
+};
+
+export const addContentCreator = async (url: string, email?: string) => {
+  const prisma = await openConnection();
+  try {
+    const { name, href, favicon, feeds } = await getBlogData(url);
+    return prisma.blog.create({
+      data: {
+        name,
+        href,
+        favicon,
+        rss: feeds[0].url, // @to-do Ustalić, który jest najlepszy
+        slug: Slugify(name, { lower: true }),
+        lastUpdateDate: NEVER,
+      },
+    });
+  } finally {
+    await closeConnection();
+  }
+};
+
+const getBlogData = async (url: string): Promise<BlogData> => {
+  const youtubeRss = getYoutubeRss(url);
+  if (youtubeRss) {
+    return {
+      name: 'Some Youtube Channel', // @to-do Ustalić, nazwę strony z yt
+      href: url,
+      favicon: 'https://www.youtube.com/s/desktop/d743f786/img/favicon.ico',
+      feeds: [
+        {
+          type: '',
+          url: youtubeRss,
+          title: youtubeRss,
+        },
+      ],
+    };
+  } else {
+    const response = await fetch(url);
+    const htmlText = await response.text();
+    if (htmlText) {
+      const $ = cheerio.load(htmlText);
+      const blogName = getBlogName($);
+      const favicon = getFavicon(url, $);
+      const feeds = searchFeed(url, $);
+      return {
+        name: blogName,
+        href: url,
+        favicon,
+        feeds,
+      };
+    }
+  }
+  throw Boom.badData();
+};
+
+const getYoutubeRss = (url: string) => {
+  const regex = /^(http(s)?:\/\/)?((w){3}.)?youtu(be|.be)?(\.com)?\/.+/gm;
+  const isYoutubeUrl = regex.test(url);
+
+  if (isYoutubeUrl) {
+    return getYoutubeChannel(url);
+  } else {
+    return false;
+  }
+};
+
+const getYoutubeChannel = (url: string) => {
+  const urlWithoutParams = url.split('?')[0];
+  if (urlWithoutParams.split('channel/')[1]) {
+    return `https://www.youtube.com/feeds/videos.xml?channel_id=${
+      urlWithoutParams.split('channel/')[1].split('/')[0]
+    }`;
+  } else if (url.split('user/')[1]) {
+    return `https://www.youtube.com/feeds/videos.xml?user=${
+      urlWithoutParams.split('user/')[1].split('/')[0]
+    }`;
+  }
+  return false;
```

Comment:
O to się już chyba nano pytała, ale dodam do review - `false` nie jest tu odpowiednie. Zwłaszcza że w kolejnej funkcji już zwracasz `null` ;)